OrbitGl/GlCanvas.cpp

```diff
@@ -352,7 +352,8 @@ float GlCanvas::ScreenToWorldWidth(int width) const {
 Vec2 GlCanvas::WorldToScreen(Vec2 world_pos) const {
   Vec2 screen_pos;
   screen_pos[0] = floorf((world_pos[0] - world_top_left_x_) / world_width_ * GetWidth());
-  screen_pos[1] = floorf((world_top_left_y_ - world_pos[1]) / world_height_ * GetHeight());
+  screen_pos[1] =
+      floorf((GetHeight() - (world_top_left_y_ - world_pos[1])) / world_height_ * GetHeight());
```

Comment:
So this is moving the origin from top-left to bottom-left, right?

This is probably a stupid question, but how has this not affected other code so far? Was WorldToScreen unused?
(I'm sorry I'm lacking a bit of context hear. :sweat_smile: )

One remark regarding the code: It's rather unrelated to your change but affects this line. We should replace the C-function `floorf` by its C++ equivalent `std::floor`. (Can also be fixed somewhen else)

Reply:
Yes. We have 2 different coordinate systems, "screen space" and "world space". I'm not 100% sure about I'm talking below, correct me if I am wrong.

Screen space refers to point in the screen. Makes sense when we want to draw items that don't have to move when we navigate between events. They are centered in the bottom left point of the screen (0,0).

World space is more complicated. They use the world-top-left as a 0,0. This world-top-left seems to be mostly recalculated after each zoom/jump/etc. It seems to be only useful when users navigate in the TimeGraph using the mouse. When this happens, only world-top-left is an invariant after mouse-movements.

There are 2 things that sound strange for me. 
1- Why we aren't using the same point as a reference (bottom left)
2- Why world-coordinates hasn't always the (0,0) at the top/bottom-leftmost point in the capture. 

As I only trying to solve a bug that I've introduced, I didn't investigate why we are drawing on that way.

Second, AFAIR WorldToScreen was created recently for tutorials (is only used there), so I tagged @reichlfl for that.

And third, thanks for the floorf! :)

Reply:
Ah great. Since WorldToScreen is new, then it makes a lot of sense.

Reply:
Actually `WorldToScreen` was introduced for the accessibility changes, and this change will break the accessibility.

It's even more complicated than explained above, because what is referred to as "ScreenSpace" in methods like `DrawScreenSpace` is *not* was this method refers to as "ScreenSpace"... we have this weird space called screen space that is in pixel coordinates, but with the bottom left being (0, 0), which is used to draw the elements "around" the capture (like scrollbars, time bar etc).

But this method is the inverse of `ScreenToWorld`, which converts from the QT screen coordinate system to our world... this is used to e.g. convert mouse coordinates from QT to OpenGl. `ScreenToWorld` thus converts from OpenGl world to QT screen, and needs to stay as it is. You'll need to introduce a new method to convert to the space you need here. Ideally we would change the "Screen" space we use in OpenGl to actually be the same as QT Screen space, but this is more effort.